### ISSUE TITLE
When loading the sl3 package, this error gets reported...

### ISSUE THREAD

**Issue by NinoMalekovic** [jeremyrcoyle/sl3#397]
> remotes::install_github("tlverse/sl3")
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo tlverse/sl3@HEAD
✔  checking for file ‘/tmp/Rtmp6QcgX6/remotes289a5a3e5f3a/tlverse-sl3-6544257/DESCRIPTION’ ...
─  preparing ‘sl3’:
✔  checking DESCRIPTION meta-information ...
─  installing the package to process help pages
         -----------------------------------
─  installing *source* package ‘sl3’ ...
   ** using staged installation
   ** R
   ** data
   *** moving datasets to lazyload DB
   ** inst
   ** byte-compile and prepare package for lazy loading
   Error in dyn.load(file, DLLpath = DLLpath, ...) : 
     unable to load shared object '/usr/local/lib/R/site-library/igraph/libs/igraph.so':
     libglpk.so.40: cannot open shared object file: No such file or directory
   Calls: <Anonymous> ... asNamespace -> loadNamespace -> library.dynam -> dyn.load
   Execution halted
   ERROR: lazy loading failed for package ‘sl3’
─  removing ‘/tmp/RtmpXVNh4Q/Rinst295d29fd7e44/sl3’
         -----------------------------------
   ERROR: package installation failed
Error: Failed to install 'sl3' from GitHub:
  ! System command 'R' failed

**Comment by szhorvat** [comment]
What is your OS?

You should test if igraph works in a fresh R session (`library(igraph)`). If not, please refer to the troubleshooting instructions: https://github.com/igraph/rigraph/wiki/Installation-Troubleshooting

**Comment by rachaelvp** [comment]
Hi @NinoMalekovic I think this might be an issue with your version of R. Try updating to the latest release, I believe that’s version 4.2. 
If that doesn’t work, or if you already are using R 4.2, then try the igraph solution mentioned above.

**Comment by szhorvat** [comment]
Actually, it's still not clear to me how this issue can arise at all. Thus, if you could answer the two questions [here](https://github.com/igraph/rigraph/issues/209#issuecomment-1273400011), that would be quite useful.

**Comment by rachaelvp** [comment]
Hi @NinoMalekovic could you please share the output to calling `sessionInfo()`? 

My first thought is that this is an issue with your version of R. I’ve seen an error like this before and updating R resolved it. Try updating to the latest release if you are not using that. I believe that’s version 4.2. 

If this doesn’t work, or if you already are using R 4.2, please let us know.

**Comment by NinoMalekovic** [comment]
First, Thank you both szhorvat and rachaelvp for responding. 

My OS is Ubuntu 20.04.5 LTS. igraph does not work in a fresh R session, and I'll try and troubleshoot it referring to: https://github.com/igraph/rigraph/wiki/Installation-Troubleshooting.

In the meanwhile, I tried installing igraph separately. This was my most recent attempt to resolve the issue. Specifically, I cannot install igraph locally. This is the error I get:

> library(igraph)
Error: package or namespace load failed for ‘igraph’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/usr/local/lib/R/site-library/igraph/libs/igraph.so':
  libglpk.so.40: cannot open shared object file: No such file or directory

However, I can install igraph on https://rstudio.cloud/, but cannot install sl3 on https://rstudio.cloud/.

"Did you by any chance uninstall GLPK after installing igraph?" No.

"Did you have a conda environment active when you installed igraph?" No. Rstudio and Python are deployed through separate Docker containers. The Docker image that we use for RStudio is retrieved from https://hub.docker.com/r/rocker/rstudio.

Before trying igraph, I had tried installing sl3 locally, as well as on https://rstudio.cloud/. The exact same error got reported in both cases. This is why I don't think this has to do with my setup.

What's really strange is the fact that a couple of weeks ago, I installed sl3 locally through the dedicated container, and it worked nicely. Specifically, not only did it work, but I also didn't have to use Github PAT to overcome the limited Github API requests. Specifically, a couple of weeks ago sl3 worked like a charm.

The output of the sessionInfo():
> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9         compiler_4.2.1     later_1.3.0        urlchecker_1.0.1  
 [5] easypackages_0.1.0 prettyunits_1.1.1  profvis_0.3.7      remotes_2.4.2     
 [9] tools_4.2.1        digest_0.6.29      pkgbuild_1.3.1     pkgload_1.3.0     
[13] memoise_2.0.1      lifecycle_1.0.2    rlang_1.0.6        shiny_1.7.2       
[17] cli_3.4.1          rstudioapi_0.13    curl_4.3.2         fastmap_1.1.0     
[21] withr_2.5.0        stringr_1.4.1      fs_1.5.2           htmlwidgets_1.5.4 
[25] gitcreds_0.1.2     devtools_2.4.4     rprojroot_2.0.3    glue_1.6.2        
[29] data.table_1.14.2  R6_2.5.1           processx_3.7.0     sessioninfo_1.2.2 
[33] callr_3.7.1        purrr_0.3.5        magrittr_2.0.3     ps_1.7.1          
[37] promises_1.2.0.1   ellipsis_0.3.2     htmltools_0.5.3    usethis_2.1.6     
[41] assertthat_0.2.1   mime_0.12          xtable_1.8-4       httpuv_1.6.5      
[45] stringi_1.7.8      miniUI_0.1.1.1     cachem_1.0.6       crayon_1.5.2      
Session restored from your saved work on 2022-Oct-10 12:33:02 UTC

**Comment by szhorvat** [comment]
Thank you for the detailed response!

Can you please uninstall and re-install igraph? `remove.packages("igraph")`, then restart R, then `install.packages("igraph")`.  When you see the line `checking for glp_read_mps in -lglpk...` during installation, does it say "yes" or "no" after it?

**Comment by szhorvat** [comment]
I've updated the troubleshooting document based on the information you provided see, see #584

**Comment by szhorvat** [comment]
First of all, this problem will almost certainly be resolved if you install GLPK. On Debian/Ubuntu, `sudo apt-get install libglpk40`.

----

Based on answers I got from others, the two most likely ways to get this error on Linux are:

 - igraph was installed _from sources_ (using the `install.packages()` command) when GLPK was present, but GLPK got inadvertently removed at a later time. That will break igraph.
 - A binary version of igraph was installed, and this binary version happens to assume that GLPK is present. This might be the case for RStudio Package Manager.

I updated the [troubleshooting document](https://github.com/igraph/rigraph/wiki/Installation-Troubleshooting) with some more information (feedback is welcome). I hope this helps.
